src/utils/hydration.js

```diff
@@ -142,19 +142,85 @@ function injectRepeatedRows({ step, repeatedRowCount }) {
   };
 }
 
+const formatTrackValue = (value, type = 'col', isEmpty) => {
+  if (isEmpty && value === 'auto') value = '0'; // Default empty value
+  if (type === 'col') return `minmax(0, ${value})`;
+  return value;
+};
+
+const formatDimensionValue = (value, type = 'col') => {
+  switch (value) {
+    case 'fill':
+      return type === 'col' ? '100%' : 'fit-content';
+    case 'auto':
+      return type === 'col' ? 'auto' : 'fit-content';
+    default:
+      return parseInt(value);
+  }
+};
+
+const getTrackElementsMap = (step) => {
+  const {
+    servar_fields: fields,
+    progress_bars: progressBars,
+    images,
+    texts,
+    buttons
+  } = step;
+
+  const allElements = [
+    ...fields,
+    ...progressBars,
+    ...images,
+    ...texts,
+    ...buttons
+  ];
+
+  const row = {};
+  const column = {};
+
+  allElements.forEach((e) => {
+    const { row_index: rowIndex, column_index: columnIndex } = e;
+
+    if (!(rowIndex in row)) row[rowIndex] = [];
```

Comment:
could you just store true or false instead of the entire array? then you can just do one line to set the index to true instead of the if check

Reply:
That would change the nature of this function, but sure, I guess it isn't needed as such at the moment.